cmd/fpga_crihook/main.go

```diff
@@ -15,420 +15,339 @@
+	} `json:"mounts"`
+	Hooks struct {
+		Prestart []struct {
+			Path string   `json:"path"`
+			Args []string `json:"args"`
+			Env  []string `json:"env"`
+		} `json:"prestart"`
+	} `json:"hooks"`
+	Annotations struct {
+		IoKubernetesContainerHash                     string    `json:"io.kubernetes.container.hash"`
+		IoKubernetesContainerName                     string    `json:"io.kubernetes.container.name"`
+		IoKubernetesContainerRestartCount             string    `json:"io.kubernetes.container.restartCount"`
+		IoKubernetesContainerTerminationMessagePolicy string    `json:"io.kubernetes.container.terminationMessagePolicy"`
+		IoKubernetesCriOContainerID                   string    `json:"io.kubernetes.cri-o.ContainerID"`
+		IoKubernetesCriOContainerType                 string    `json:"io.kubernetes.cri-o.ContainerType"`
+		IoKubernetesCriOCreated                       time.Time `json:"io.kubernetes.cri-o.Created"`
+		IoKubernetesCriOLogPath                       string    `json:"io.kubernetes.cri-o.LogPath"`
+		IoKubernetesCriOMetadata                      string    `json:"io.kubernetes.cri-o.Metadata"`
+		IoKubernetesCriOSandboxID                     string    `json:"io.kubernetes.cri-o.SandboxID"`
+		IoKubernetesCriOSandboxName                   string    `json:"io.kubernetes.cri-o.SandboxName"`
+		IoKubernetesCriOSeccompProfilePath            string    `json:"io.kubernetes.cri-o.SeccompProfilePath"`
+		IoKubernetesCriOStdin                         string    `json:"io.kubernetes.cri-o.Stdin"`
+		IoKubernetesCriOStdinOnce                     string    `json:"io.kubernetes.cri-o.StdinOnce"`
+		IoKubernetesCriOTTY                           string    `json:"io.kubernetes.cri-o.TTY"`
+		IoKubernetesPodName                           string    `json:"io.kubernetes.pod.name"`
+		IoKubernetesPodNamespace                      string    `json:"io.kubernetes.pod.namespace"`
+		IoKubernetesPodTerminationGracePeriod         string    `json:"io.kubernetes.pod.terminationGracePeriod"`
+		IoKubernetesPodUID                            string    `json:"io.kubernetes.pod.uid"`
+	} `json:"annotations"`
+	Linux struct {
+		Resources struct {
+			Devices []struct {
+				Allow  bool   `json:"allow"`
+				Access string `json:"access"`
+				Type   string `json:"type,omitempty"`
+				Major  int    `json:"major,omitempty"`
+				Minor  int    `json:"minor,omitempty"`
+			} `json:"devices"`
+			Memory struct {
+				Limit int `json:"limit"`
+			} `json:"memory"`
+			CPU struct {
+				Shares int `json:"shares"`
+				Quota  int `json:"quota"`
+				Period int `json:"period"`
+			} `json:"cpu"`
+			Pids struct {
+				Limit int `json:"limit"`
+			} `json:"pids"`
+		} `json:"resources"`
+		CgroupsPath string `json:"cgroupsPath"`
+		Namespaces  []struct {
+			Type string `json:"type"`
+		} `json:"namespaces"`
+		Devices           []Device `json:"devices"`
+		RootfsPropagation string   `json:"rootfsPropagation"`
+	} `json:"linux"`
```

Comment:
can you move most of those OCI related parsing somewhere to `pkg/*` ? e.g. `pkg/oci-hooks` or something?